solr/solrj/src/test/org/apache/solr/client/solrj/impl/Http2SolrClientTest.java

```diff
@@ -469,6 +474,43 @@ public void testGetById() throws Exception {
     }
   }
 
+  @Test
+  public void testGetByIdArrayBlockingQueue() throws Exception {
+    DebugServlet.clear();
+
+    // client setup needs to be same as HttpShardHandlerFactory
+
```

Comment:
ah..... okay, so you are seeing an existing problem.  I would have recommended modifying an existing distributed-search test.  Your new test is fine though.

Reply:
it's not strictly needed anymore, was only trying to reproduce the deadlock. I can remove it if it doesn't fit with the rest of the tests